### PR TITLE
doc: add inter-vm communication example

### PR DESCRIPTION
**Pull request by liuyuan-GitHub**
Add the example for two VMs communication

Signed-off-by: Yuan Liu <yuan1.liu@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 1]
```diff
@@ -111,3 +111,59 @@ where
...
+
+  modprobe uio
+  modprobe uio_pci_generic
+  echo "1af4 1110" > /sys/bus/pci/drivers/uio_pci_generic/new_id
+
+Finally, user application can get shared memory base address from ``ivshmem`` device
+BAR resource (/sys/class/uio/uioX/device/resource2) and shared memory size from
+``ivshmem`` device config resource (/sys/class/uio/uioX/device/config).
+
+.. note:: uioX can get from one command
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:166]: Is what you mean here that the `X` in `uioX` is a number that can be retrieved using the `ls` command below?

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 2]
```diff
@@ -111,3 +111,59 @@ where
...
+  $vm_name
+
+Secondly, boot two VMs and use ``lspci`` to check the virtual device is ready for each VM.
+
+-  For VM1, it is ``00:06.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+-  For VM2, it is ``00:05.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+
+Thirdly, using below commands to probe the device::
+
+  modprobe uio
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:158]:
```suggestion
  $ sudo modprobe uio
```

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 3]
```diff
@@ -111,3 +111,59 @@ where
...
+  modprobe uio_pci_generic
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:159]:
```suggestion
  $ sudo modprobe uio_pci_generic
```

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 4]
```diff
@@ -111,3 +111,59 @@ where
...
+  echo "1af4 1110" > /sys/bus/pci/drivers/uio_pci_generic/new_id
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:160]:
```suggestion
  $ sudo echo "1af4 1110" > /sys/bus/pci/drivers/uio_pci_generic/new_id
```

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 5]
```diff
@@ -111,3 +111,59 @@ where
...
+
+Finally, user application can get shared memory base address from ``ivshmem`` device
+BAR resource (/sys/class/uio/uioX/device/resource2) and shared memory size from
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:163]:
```suggestion
BAR resource (``/sys/class/uio/uioX/device/resource2``) and shared memory size from
```

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 6]
```diff
@@ -111,3 +111,59 @@ where
...
+``ivshmem`` device config resource (/sys/class/uio/uioX/device/config).
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:164]:
```suggestion
``ivshmem`` device config resource (``/sys/class/uio/uioX/device/config``).
```

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 1 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:166]: Yes, I will refine the statement in here.

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 2 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:158]: Yes, it needs sudo

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 3 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:159]: OK

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 4 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:160]: Yes, needs sudo in here

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 5 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:163]: Got it, thanks

File: doc/developer-guides/hld/ivshmem-hld.rst (hunk 6 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/ivshmem-hld.rst:164]: Got it, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/developer-guides/hld/ivshmem-hld.rst [hunk 7]
```diff
@@ -111,3 +111,59 @@ where
...
+  $vm_name
+
+Secondly, boot two VMs and use ``lspci`` to check the virtual device is ready for each VM.
+
+-  For VM1, it is ``00:06.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+-  For VM2, it is ``00:05.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+
+Thirdly, using below commands to probe the device::
+
+  sudo modprobe uio
```
Comment (gvancuts) [inline doc/developer-guides/hld/ivshmem-hld.rst:158]: We typically all icnldue the `$` symbol in these code-blocks. But perhaps this is something that @dbkinder will do as part of his review/editing?

**Comment by wenlingz** [comment]
@liuyuan-GitHub make doc fail, please refer log:
make -C /root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/doc html BUILDDIR=/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc
make[1]: Entering directory '/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/doc'

New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc/rst/developer-guides/hld/ivshmem-hld.rst:129: WARNING: Literal block expected; none found.
/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc/rst/developer-guides/hld/ivshmem-hld.rst:143: WARNING: Literal block expected; none found.
/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc/rst/developer-guides/hld/ivshmem-hld.rst:168: WARNING: Literal block expected; none found.

make[1]: Leaving directory '/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/doc'
make[1]: *** [Makefile:61: html] Error 1
make: *** [Makefile:207: doc] Error 2
14:47:02 LOG: sshpass -p 123456 ssh -o Str

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**